src/agoradatatools/etl/transform/rna_de_aggregate.py

```
@@ -230,9 +230,11 @@ def transform_rna_de_aggregate(
             age_entries = {}
             for row in group.itertuples(index=False):
                 age = str(row.age)
+                # Adding 0.0 to avoid -0.0 values
+                # Adding zero to negative zero produces positive zero according to IEEE 754 floating-point arithmetic rules
                 age_entries[age] = {
-                    "log2_fc": float(row.log2foldchange),
-                    "adj_p_val": float(row.padj),
+                    "log2_fc": float(row.log2foldchange) + 0.0,
```

I think this works. But I would recommend a cleaner approach: 
```
def normalize_zero(value: float) -> float:
    """Convert -0.0 to 0.0 while preserving other values."""
    return 0.0 if value == 0.0 else value

# Then use it:
age_entries[age] = {
    "log2_fc": normalize_zero(float(row.log2foldchange)),
    "adj_p_val": 0.0 if pd.isna(row.padj) else normalize_zero(float(row.padj)),
}
```

---

I don't understand what normalize_zero does or how it catches -0.0 since you're checking equality with +0.0. 
Further, for p-values, they can't be negative so you just need to use np.fabs (floating point abs value):
```
"adj_p_val": 0.0 if pd.isna(row.padj) else np.fabs(row.padj)
```
I *think* that should work but you should test it.

---

Hmm actually, while np.fabs does do what I expect, this raises the issue that any _actually_ negative p-values in the data would get converted to positive values too. There _shouldn't_ ever be any p-values < 0 in the data, and if there are that should raise an error somewhere.  We should either check for negative values here before using np.fabs(), OR use Lingling's suggestion and let GX catch negative values. Not sure which approach is cleaner.